Review chapter 2 (end of review at "Variabili user defined" (#6)
d3nzl/SystemAdministration: LaTeX/ShellScripting.tex

In questo contesto risulta particolarmente utile includere definizioni di variabili e/o funzioni presenti in un determinato script, all'interno di un altro script, così da consentirne il riutilizzo.

A tal proposito il comando \texttt{source} , paragonabile alla direttiva \texttt{\#include}, permette di eseguire un file script nello stesso ambiente d'esecuzione del comando stesso, consentendo di creare vere e proprie librerie di funzioni o anche set di variabili in grado di essere importate e riutilizzate in base alle necessità dettate dal contesto.

\paragraph{Portabilità}
È possibile specificare un interprete in maniera tale che lo script risulti più facilmente portabile fra ambienti con caratteristiche differenti; a tal scopo si ricorre all'utility \texttt{env}, che attraverso le variabili d'ambiente identifica la posizione e la versione nel sistema dell'interprete da eseguire; la sintassi sarà del formato \texttt{\#!/usr/bin/env bash}.

\subsubsection{Variabili d'ambiente}

> Vi sono alcuni dati, solitamente riguardanti il sistema o le preferenze relative ad uno specifico utente, che sono utili a una varieta' di comandi (es. la versione del sistema operativo, la lingua dell'utente, etc..); a tal proposito, UNIX mette a  disposizione le \textbf{variabili d'ambiente}, in modo che tutte queste informazioni siano disponibili ed accessibili da un eventuale processo in esecuzione su una shell.\newline

Reviewer: ```suggestion
Vi sono alcuni dati, solitamente riguardanti il sistema o le preferenze relative ad uno specifico utente, che sono utili a una varieta' di comandi (es. la versione del sistema operativo, la lingua dell'utente, etc..); a tal proposito, UNIX mette a  disposizione le \textbf{variabili d'ambiente}, per mezzo delle quali queste informazioni vengono rese disponibili ed accessibili da un eventuale processo in esecuzione su una shell.\newline
```

\subsection{Shell Scripting Basics}
\subsubsection{Introduzione}
Uno shell script è un file di testo con impostata la flag di esecuzione, che inizia con la stringa \texttt{\#!/bin/sh} o \texttt{\#!/bin/bash}. Tale stringa viene vista dalla shell stessa come un \textbf{commento} (come ogni stringa preceduta dal \texttt{\#}), 
mentre il sistema operativo per la sintassi \texttt{\#!} identifica
il \emph{magic number} degli script, ovvero file contenenti sequenze di comandi eseguite con l'ausilio di un \textbf{interprete}.
\newline \texttt{/bin/bash} rappresenta l'interprete da invocare per gli script shell, ma il meccanismo è applicabile a tutti gli altri linguaggi (ad es. \texttt{\#!/usr/bin/perl}).\\
I caratteri contenuti in uno script vengono interpretati dalla shell come se fossero digitati dall'utente in quel preciso istante.
\newline 
La possibilità di codificare in uno script comandi anche complessi richiede la disponibilità di strutture di controllo per eseguire test o per iterare azioni ripetitive. 
Tutte le strutture accettate da uno script possono essere utilizzare sulla linea di comando. 

> % EH? (parlo della riga sotto)

Author: Qua ho fatto un commento che dovresti vedere

consentono inoltre di salvare informazioni temporanee che saranno facilmente accessibili
da qualsiasi programma in esecuzione sulla shell, fornendo un meccanismo ausiliario
per salvare
in maniera persistente dati necessari.\\
Ci sono due tipologie di variabili d'ambiente nella shell bash:
\begin{itemize}
	\item{Variabili \textbf{Globali}}
	\item{Variabili \textbf{Locali}}
\end{itemize}
\paragraph{Variabili Globali}
Le variabili \textbf{globali} sono visibili dalla sessione di shell corrente
e vengono ereditate da qualsiasi processo figlio (subshell) generato; contrariamente,
le variabili \textbf{locali} rimangono visibili unicamente alla shell che le crea.
Le variabili globali sono particolarmente utili per applicazioni che necessitano di condividere
delle informazioni con i processi figli.\newline
Il sistema linux imposta diverse variabili d'ambiente globali ad ogni esecuzione di una
sessione shell. Le variabili relativa ad informazioni di sistema si trovano generalmente
in forma uppercase, per differenziarle da quelle utilizzate dall'utente.
Per visualizzare le variabili globali è possibili utilizzare i comandi \texttt{env}
o \texttt{printenv}.
\paragraph{Variabili Locali}

> Le variabili locali, come specificato dal nome, sono accessibili unicamente
> dal processo locale in cui vengono definite. Non esiste un comando che mostri
> unicamente le variabili locali; è possibile invece utilizzare \texttt{set}, che riporta 
> in output tutte le variabili definite per uno specifico processo, ovvero locali,
> globali e quelle definite dall'utente.

Reviewer: ```suggestion
Le variabili locali, come specificato dal nome, sono accessibili unicamente dal processo locale in cui vengono definite. Non esiste un comando che mostri unicamente le variabili locali; è possibile invece utilizzare \texttt{set}, che riporta in output tutte le variabili definite per uno specifico processo, ovvero locali, globali e quelle definite dall'utente.
```

\paragraph{Variabili user-defined}

> Una volta eseguito un processo shell, è possibile definire nuove variabili (user-defined).
> È possibile assegnare sia valori numerici che stringe ad una variabile d'ambiente utilizzando
> l'operatore d'assegnamento \texttt{=}, facendo attenzione che non siano presenti spazi prima
> e dopo l'uguale.

Reviewer: ```suggestion
Una volta eseguito un processo shell, è possibile definire nuove variabili (user-defined); queste possono assumere sia valori numerici che stringhe, e vengono associate utilizzando l'operatore d'assegnamento \texttt{=}, facendo attenzione che non siano presenti spazi prima e dopo l'uguale.
```

 \texttt{\#!/usr/bin/env bash}.

\subsubsection{Variabili d'ambiente}

> Vi sono alcuni dati, solitamente riguardanti il sistema o le preferenze relative ad uno specifico utente, che sono utili a una varieta' di comandi (es. la versione del sistema operativo, la lingua dell'utente, etc..); per questo motivo UNIX mette a disposizione un meccanismo per far si che tutti queste informazioni siano disponibili ad un eventuale processo che le necessitasse: le \textbf{variabili d'ambiente}.\newline

Author: Oh no, qua c'è qualcosa che non va

Author: Ma, scusa, dire

>  Esistono alcuni dati, solitamente riguardanti il sistema o le preferenze relative ad uno specifico utente, che sono utili a una varietà di comandi (es. la versione del sistema operativo, la lingua dell'utente, etc..); UNIX mette a disposizione le \emph{variabili ambiente}, in modo che queste informazioni
> siano sempre disponibili per un eventuale processo in esecuzione su una shell

 non è meglio?

Reviewer: ```suggestion
Vi sono alcuni dati, solitamente riguardanti il sistema o le preferenze relative ad uno specifico utente, che sono utili a una varieta' di comandi (es. la versione del sistema operativo, la lingua dell'utente, etc..); a tal proposito, UNIX mette a  disposizione le \textbf{variabili d'ambiente}, in modo che tutte queste informazioni siano disponibili ed accessibili per un eventuale processo in esecuzione su una shell.\newline
```

> Queste vengono utilizzare dalla shell per salvare informazioni riguardanti la sessione corrente e l'ambiente di lavoro; consentono inoltre di salvare informazioni temporanee che saranno facilmente accessibili da qualsiasi programma in esecuzione sulla shell, fornendo quindi un meccanismo ausiliario per salvare in maniera persistente dati necessari.\\

Reviewer: ```suggestion
Queste vengono utilizzate dalla shell per mantenere informazioni inerenti la sessione corrente e l'ambiente di lavoro; consentono inoltre di archiviare informazioni temporanee che saranno poi facilmente accessibili da qualsiasi programma in esecuzione sulla shell, fornendo quindi un meccanismo ausiliario per immagazzinare in maniera persistente dati necessari al corretto funzionamento del programma.\\
```

	\item{Variabili \textbf{Locali}}
\end{itemize}
\paragraph{Variabili Globali}

> Le variabili cosidette \textbf{globali} sono visibili dalla sessione di shell corrente e vengono ereditate da qualsiasi processo figlio (subshell) generato; contrariamente, le variabili \textbf{locali} rimangono visibili unicamente alla shell che le crea. Questo rende le variabili globali particolarmente utili per applicazioni che necessitano di condividere delle informazioni con i processi figli.\newline

Author: Io avevo tolto "cosidette" perchè sono dell'idea che meno parole usi e più chiaro è; considera inoltre che le variabili sono globali, e non è un termine "informale".

Reviewer: Giusto giusto, mi e' scappato riportando tutto su una riga
```suggestion
Le variabili \textbf{globali} sono visibili dalla sessione di shell corrente e vengono ereditate da qualsiasi processo figlio (subshell) generato; contrariamente, le variabili \textbf{locali} rimangono visibili unicamente alla shell che le crea. Questo rende le variabili globali particolarmente utili per applicazioni che necessitano di condividere delle informazioni con i processi figli.\newline
```

> Il sistema linux imposta diverse variabili d'ambiente globali ad ogni esecuzione di una sessione shell. Le variabili relative ad informazioni di sistema si trovano generalmente in forma uppercase, per differenziale da quelle utilizzate dall'utente. Per visualizzare le variabili globali e' possibili utilizzare i comandi \texttt{evn} o \texttt{printenv}.

Author: Ti farò una review dove fixerò solo gli accenti :)
